### ISSUE TITLE
Motion Component Breaks Parent's setState

### ISSUE THREAD

**Issue by d4tocchini** [infernojs/inferno-motion#1]
When `Overlay` renders a `Motion` component, then `Demo` throws `TypeError: Cannot read property 'dom' of null` on `setState`.

```javascript
import Inferno from 'inferno'
import Component from 'inferno-component'
import {fromJS, raf} from '@the-grid/deps'
import {Motion, spring} from 'inferno-motion';

let spring_default = {size: 1, o: 1}
let spring_show = {size: spring(1), o: spring(1)}
let spring_hide = {size: spring(1), o: spring(0)}

class Demo extends Component {
  state = {shown: true}
  onclick = (e) => {
    e.preventDefault();
    this.show()
  }
  show() {
    this.setState({shown: !this.state.shown})
  }
  render() {
    var shown = this.state.shown
    return (
      <div>
        <Overlay show={shown} />
        <a id="show-button" onClick={this.onclick} style={{position:'fixed',top:0,left:0}}>show</a>
      </div>
    )
  }
}

function Overlay(props) {
  let {
    show
  } = props
  let scope = "#thing-i-want-to-style"
  let style = show ? spring_show : spring_hide
  return (
    <div>
      <Motion defaultStyle={spring_default} style={style} >{
        (sprung)=>{
          let css =  `
            ${scope} {
              visibility: ${sprung.o <= 0 ? 'hidden' : 'inherit'} !important;
              opacity: ${sprung.o};
              transform: scale(${sprung.size});
            }
          `
          return (
            <style dangerouslySetInnerHTML={{__html: css}} />
          )
        }
      }</Motion>
    </div>
  )
}
```

![screenshot 2016-12-21 03 15 37](https://cloud.githubusercontent.com/assets/26006/21387152/31202b8a-c72b-11e6-8f46-f7a7dbfd46f6.png)

**Comment by Havunen** [comment]
Thanks for reporting this issue. I think it will be fixed in future Inferno release 1.0.6?

**Comment by Havunen** [comment]
This has been fixed.
